spec/requests/api/services_spec.rb

```diff
@@ -625,4 +625,58 @@ def expect_svc_with_vms
       end
     end
   end
+
+  describe 'Orchestration Stack subcollection' do
+    let(:os) { FactoryGirl.create(:orchestration_stack) }
+
+    before do
+      svc.add_resource!(os, :name => ResourceAction::PROVISION)
+    end
+
+    it 'can query orchestration stacks as a subcollection' do
+      api_basic_authorize subcollection_action_identifier(:services, :orchestration_stacks, :read, :get)
+
+      run_get("#{services_url(svc.id)}/orchestration_stacks", :expand => 'resources')
+
+      expected = {
+        'resources' => [
+          a_hash_including('id' => os.id)
+        ]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+
+    it 'can query a specific orchestration stack' do
+      api_basic_authorize subcollection_action_identifier(:services, :orchestration_stacks, :read, :get)
+
+      run_get("#{services_url(svc.id)}/orchestration_stacks/#{os.id}")
+
+      expected = {
+        'id' => os.id
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+
+    it 'will not return orchestration stacks without an appropriate role' do
+      api_basic_authorize
+
+      run_get("#{services_url(svc.id)}/orchestration_stacks")
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'will return an empty array for a service without orchestration_stacks' do
+      api_basic_authorize subcollection_action_identifier(:services, :orchestration_stacks, :read, :get)
+
+      run_get("#{services_url(svc1.id)}/orchestration_stacks", :expand => 'resources')
+
+      expected = {
+        'resources' => []
```

Comment:
Is this because `service.orchestration_stacks # => []` Or because `service.orchestration_stacks # => NoMethodError`? It looks like the former from what I understand, so wondering if the `respond_to?` check is needed above?